clang/test/CIR/CallConvLowering/AArch64/aarch64-cc-structs.c

```diff
@@ -73,3 +73,28 @@ GT_128 ret_gt_128() {
   GT_128 x;
   return x;
 }
+
+// CHECK: cir.func {{.*@pass_lt_64}}(%arg0: !u64
+// CHECK:   %[[#V0:]] = cir.alloca !ty_LT_64_, !cir.ptr<!ty_LT_64_>
+// CHECK:   %[[#V1:]] = cir.cast(integral, %arg0 : !u64i), !u16i
+// CHECK:   %[[#V2:]] = cir.cast(bitcast, %[[#V0]] : !cir.ptr<!ty_LT_64_>), !cir.ptr<!u16i>
```

Comment:
This is really neat :) I'm trying to follow along some of the ABI work, and I noticed that OG does a `getelementptr` instead of a bitcast: https://godbolt.org/z/6bv5ns9Pq. They should be totally equivalent in this case, but I was wondering where that divergence came from and what the reason for it was. (Sorry if you or others have already explained that elsewhere; you can just point me to it if so.)

Reply:
Looking at the bitcast handling in `LowerToLLVM.cpp:1148`, seems like we unconditionally generate a `mlir::LLVM::BitcastOp`, but perhaps we should specialize further and emit GEPs when unwrapping structs like this?

Reply:
Well, `gep` is emitted [here](https://github.com/llvm/clangir/blob/main/clang/lib/CodeGen/CGCall.cpp#L1191) in the original codeden, but we don't do it so far  - we have a missing feature guard

Frankly speaking, I have no idea, why do we need gep and get the address of the first field here, i.e. how it differs from the pointer to a struct itself. But we could use `GetMemberOp` instead - and thus we'll get gep in the LLVM IR.

The bitcasts disappear in the LLVM IR - probably because all of them are  for pointers cast which is useless for opaque pointers.

Reply:
I filed an issue for us to keep track of this